modules/rtpengine/rtpengine.c

```diff
@@ -1375,6 +1397,11 @@ mod_init(void)
 		memset(&dlgb, 0, sizeof(struct dlg_binds));
 	}
 
+  if (register_timer("rtpengine-timer", rtpengine_timer, NULL, rtpengine_timer_interval, TIMER_FLAG_SKIP_ON_DELAY) <0 ) {
```

Comment:
Thanks for the quick review!

I wasn't sure if the `TIMER_FLAG_DELAY_ON_DELAY` flag would cause issues in the case where you have several unreachable nodes for an extended period of time.  Say with just 2 down nodes and default module params, the probing would take 2 nodes * 1 sec timeout * 5 retries = 10sec to complete.  Since task time > time interval (5 sec by default) wouldn't this lead to a uncontrollable buildup of timer tasks?  I'm not familiar with the mechanics of the timer manager, so maybe this case is handled.

Reply:
A timer job is handled by *any* free OpenSIPS process. This means that in your scenario with 2 nodes down, only one process will be stuck for 10s, but any other (free) node can take any of the other timer jobs.
The only downside is that if the job takes too long, for a very long time, all those jobs might stack up - perhaps it is a good idea to limit the number of jobs stacked, but that's probably a different discussion.

Reply:
Hi folks, there is no need to worry about jobs stacking, as both `SKIP_ON_DELAY` and `DELAY_ON_DELAY` schedule a maximum of 1 concurrent timer job executions (see [here](https://github.com/OpenSIPS/opensips/blob/master/timer.c#L339)).

Reply:
Okay, perfect. I will push the `DELAY_ON_DELAY` change in bit. Thanks!